client/src/app/s-degree-plan-editor/s-degree-plan-editor.component.ts

```diff
@@ -30,9 +30,12 @@ export class SDegreePlanEditorComponent implements OnInit {
       this.plan.user = "";
       this.plan.program = "";
       this.plan.isAddingYear = false;
-      this.plan.years.forEach(
-        (semester) => (semester.isAddingSemester = false)
-      );
+      this.plan.years.forEach((year) => {
```

Comment:
I realized that I need to change isAddingSemester into an object that contains both a boolean flag (for toggling the add semester widget) and then a string property to hold the value of the select menu of terms.